### ISSUE TITLE
Edited & normal message updates simultaneously

### ISSUE THREAD

**Issue by iG0tB0lts** [python-telegram-bot/python-telegram-bot#588]
Using the python-telegram-bot wrapper, I'm trying to have my bot print some text if it receives a message, then print some other text if it's edited.
However I can only have 1 of the following: if I use
`update.message.reply_text("Original")`
It prints 'Original' successfully. However if bot receives a msg with the following code:
```
update.edited_message.reply_text("Edited")
update.message.reply_text("Original")
```
It does not print 'Original' but will only print 'Edited' once the message is edited.

If I put the `update.message.reply_text` first, it will print "Original" but will not print "Edited" when the message is edited

So basically whichever statement is first, only that is executed. Is there anyway around this? Thanks any help would be greatly appreciated

P.S here is my full code https://pastebin.com/mS6Y98x1

**Comment by iG0tB0lts** [comment]
Alright so I fixed it by creating 2 functions, then put the dispatch handler of the edit function after the normal message function.

**Comment by Eldinnie** [comment]
an `update` can contain either `message` OR `edited_message` never both.

**Comment by bomjacob** [comment]
This could probably be able by doing something like
``` python
msg = update.message or update.edited_message
msg.reply_text("Original")
```
And then having allow_edited=True on your handler.

**Comment by Eldinnie** [comment]
@bomjacob did you manage to reproduce this faulty behaviour?

**Comment by bomjacob** [comment]
Okay, I know what's going on now. It's due to edited_message being None when it's a new message, and since that will raise an exception it will not continue on to the next line. 
I'm not entirely sure what you're trying to acceive @iG0tB0lts, but after reading your post once again, it seems that the [code above](https://github.com/python-telegram-bot/python-telegram-bot/issues/588#issuecomment-300280517) might actuallly be what you want.

**Comment by iG0tB0lts** [comment]
Hi, thanks for the reply, sorry I wasn't monitoring this thread.
I came up with a solution like I mentioned in my comment, basically it was the order of dispatch handler, i.e
` dp.add_handler(MessageHandler(Filters.text, save_msg))
  dp.add_handler(MessageHandler(Filters.text, see_edit, allow_edited=True))`

If I put the see_edit function handler first, it results in the above issue.

Thanks for your code, I will try that out as well, seems like an efficient method(I'm a total beginner in python)
I guess you could mark this issue as solved? (Beginner with github also :P)
